test/test-viz.cpp

```diff
@@ -0,0 +1,311 @@
+            for(auto ins : vert.ins) {
+                out << scheme[ins].name << ' ';
+            }
+            std::cout << std::endl;
+        }
+        std::cout << "\n====QUEUE===\n";
+        for(auto &requests : queue) {
+            out << std::get<0>(requests).type;
+        }
+    }
+
+    std::vector<Vertex> scheme;
+    std::vector<std::tuple<Vertex, std::string>> queue;
+    int outputNum;
+};
+
+class bench_parser : public lorina::bench_reader {
+public:
+    explicit bench_parser(logic_scheme& logicScheme) : scheme(logicScheme) {}
+
+    virtual void on_input(const std::string& name) const override {
+        scheme.add_input(name);
+    }
+
+    virtual void on_output(const std::string& name) const override {
+        scheme.add_output(name);
+    }
+
+    virtual void on_dff(const std::string &input, const std::string &output) const override {
+        scheme.add_dff(input, output);
+    }
+
+    virtual void on_gate(const std::vector<std::string>& inputs, const std::string& output, const std::string& type) const override {
+        scheme.add_gate(inputs, output, type);
+    }
+
+    logic_scheme& scheme;
+};
+
+void setPosition(std::vector<Vertex*> elements, const int max_elements, const float height, const float width, const int screen_height) {
+    int count = 0;
+    int freeSpaces = elements.size() + 1;
+    float step = (screen_height - height * elements.size()) / freeSpaces;
+    for(Vertex* elem : elements) {
+        elem->position.x = elem->layer * HSPACING * width;
+        elem->position.h = height;
+        elem->position.w = width;
+
+        if (elements.size() == max_elements) {
+            elem->position.y = count * VSPACING * height;
+        }
+        else {
+            elem->position.y = step * (count + 1) + height * count;
+        }
+
+        count++;
+    }
+}
+
+void setConnections(Vertex* element, const std::vector<Vertex>& scheme, std::vector<Line>& connections) {
+    for(int& input : element->ins) {
+        Line connect(scheme[input], *element);
+        connections.push_back(connect);
+    }
+}
+
+void drawElementsAndConnections(SDL_Renderer* renderer, const logic_scheme& elements, const std::vector<Line>& connections) {
+    SDL_SetRenderDrawColor(renderer, 255,255,255, SDL_ALPHA_OPAQUE);
+
+    //Placing elements on the screen
+    for(const Vertex& element : elements.scheme) {
+        SDL_RenderDrawRectF(renderer, &element.position);
+    }
+
+    //Placing connection lines on the screen
+    for(const Line& connection_line : connections) {
+        SDL_RenderDrawLineF(renderer, connection_line.x1, connection_line.y1, connection_line.x2, connection_line.y2);
+    }
+}
+
+void drawBackground(SDL_Renderer* renderer) {
+    SDL_SetRenderDrawColor(renderer, 0,0,0, SDL_ALPHA_OPAQUE);
+    SDL_RenderClear(renderer);
+}
+
+int main(int argc, char* argv[]) {
+
+    std::cout << "Enter the file path: ";
+    std::string filename;
+    std::cin >> filename;
```

Comment:
It is generally preferred for simple applications to pass arguments via CLI, i.e. `argv`